src/NUnitFramework/framework/Internal/Commands/FixturePerTestCaseCommand.cs

```diff
@@ -53,17 +53,15 @@ public FixturePerTestCaseCommand(TestCommand innerCommand)
 
             BeforeTest = (context) =>
             {
-                if (typeInfo != null && !typeInfo.IsStaticClass && testSuite.Fixture == null)
```

Comment:
The first Parallel test will set the testSuite then subsequent tests will not enter this block because it is already set **unless** one of the tests happens to finish first and it is set to `null` in `AfterTest`. I think this is why this works for non-parallel tests because each test sets it then sets it to null for the next test.